apps/oxlint/src/command/mod.rs

```diff
@@ -25,6 +25,31 @@ pub struct MiscOptions {
     /// Number of threads to use. Set to 1 for using only 1 CPU core
     #[bpaf(argument("INT"), hide_usage)]
     pub threads: Option<usize>,
+
+    /// This option outputs the configuration to be used. To print current configuration in your project, use `--print-config all`, print configuration for a specific file is not supported for now.
+    /// When present, no linting is performed and only config-related options are valid.
+    #[bpaf(hide_usage)]
+    pub print_config: Option<PrintConfig>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub enum PrintConfig {
+    /// Print the configuration for all files
+    #[default]
+    All,
+    Path(PathBuf),
+}
+
+impl std::str::FromStr for PrintConfig {
```

Comment:
`PathBuf` can handle non-utf8 inputs since they are valid in Windows/Linux, `FromStr` can handle only valid utf8 so doing this you break a bit of compatibility. In practice this is not a big problem, but I wouldn't do it unless there's a reason.

I'm assuming you want to treat `--print-config` as "let's check all the configs" and `--print-config foo` as "let's check the config in `foo`.

You want two parsers for each case, both parsers should produce `PrintConfig` type. IT is perfectly valid to have multiple parsers with the same name, here - `--print-config`, you want to try to run that parses a file name first. Easiest way would be this is something like this:

```
#[derive(Debug, Clone, Default, Bpaf)]
pub enum PrintConfig {
    Path(
        /// help if you want
        #[bpaf(long("print-config"))] PathBuf),

    /// Print the configuration for all files
    #[default]
    #[bpaf(long("print-config"))]
    All,
    
}
```